tests/js-custom-authorization/custom_authorization.py

```diff
@@ -378,6 +378,34 @@ def run_content_types(args):
         network = test_unknown_path(network, args)
 
 
+@reqs.description("Test request object API")
+def test_random_api(args):
+    # Test that Math.random() does not repeat values:
+    # - on multiple invocations within a single request
+    # - on multiple requests
+    # - on network restarts
+
+    seen_values = set()
+
+    def assert_fresh(n):
+        assert n not in seen_values, f"{n} has already been returned"
+        seen_values.add(n)
+
+    n_repeats = 3
+    for _ in range(n_repeats):
+        with infra.network.network(
+            args.nodes, args.binary_dir, args.debug_nodes, args.perf_nodes, pdb=args.pdb
+        ) as network:
+            network.start_and_open(args)
```

Comment:
Restarting an network seems expensive - could we simply run it on different nodes in the initial service instead?

Reply:
This is expensive, as I want to ensure we're as pessimistic as possible. The test still only takes ~13s, so I think this is fine? Testing across multiple nodes is also a good idea though - I'll add that.